### DISCUSSION TITLE
Custom compile time slang lib like `core.meta.slang`, ...

### DISCUSSION THREAD

**Discussion by fknfilewalker** [shader-slang/slang#3859] [General]
I am confused how core.meta.slang, glsl.meta.slang, ... are handled so that they are also correctly recognised by the slang language server for autocompletion (I hope they are not baked into the language server). Is it possible to compile your own slang code to something similar (Slang IR Module?) and save that e.g., inside or next to the executable and still get correct syntax highlighting when runtime slang code wants to use it? Or does this only work with a forward decl using `[__extern]`?

My goal is to write some slang code that is compiled and included during host code compilation which can then be used by my runtime slang shaders. If I am completely wrong please tell me the proper way.

Bonus question: How to use slang modules saved as a file without the .slang ending, e.g., if it is defined in a .cpp file.

**Comment by csyonghe** [comment]
Please see our documentation on module pre compilation: https://shader-slang.com/slang/user-guide/link-time-specialization.html
